### ISSUE TITLE
[Bug] Build failed on macOS 12.2

### ISSUE THREAD

**Issue by xxyzz** [vgmstream/vgmstream#1069]
I use the commands from doc/BUILD.md:

```
$ mkdir build
$ cd build
$ cmake -DBUILD_AUDACIOUS:BOOL=OFF ..
-- The C compiler identification is AppleClang 13.0.0.13000029
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- The CXX compiler identification is AppleClang 13.0.0.13000029
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PkgConfig: /usr/local/bin/pkg-config (found version "0.29.2")
-- Checking for module 'libavcodec'
--   Found libavcodec, version 59.18.100
-- Checking for module 'libavformat'
--   Found libavformat, version 59.16.100
-- Checking for module 'libavdevice'
--   Found libavdevice, version 59.4.100
-- Checking for module 'libavutil'
--   Found libavutil, version 57.17.100
-- Checking for module 'libavfilter'
--   Found libavfilter, version 8.24.100
-- Checking for module 'libswscale'
--   Found libswscale, version 6.4.100
-- Checking for module 'libpostproc'
--   Found libpostproc, version 56.3.100
-- Checking for module 'libswresample'
--   Found libswresample, version 4.3.100
Downloading libg719_decode (git)...
Downloading LibAtrac9 (git)...
Downloading celt-0061 (file)...
Downloading celt-0110 (file)...
-- Checking for module 'jansson>=2.3'
--   Found jansson, version 2.14
--  Option Summary
-- =================
--         MPEG: ON (system)
--       Vorbis: ON (system)
--       FFmpeg: ON (system)
--   MAIATRAC3+: OFF
--      G.722.1: ON (built-in)
--        G.719: ON (download)
--       ATRAC9: ON (download)
--     FSB CELT: ON (download)
--        SPEEX: ON (system)
--        LIBAO: ON (system)
--      Jansson: ON (system)
--
--         Building
-- =========================
--              CLI: ON
--     vgmstream123: ON
-- Audacious plugin: OFF
--   Static linking: OFF
--
--       Install Paths
-- =========================
-- CLI/vgmstream123: /usr/local/bin
--
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/x/git/vgmstream/build
$ make
[  0%] Generating ../version_auto.h
[  0%] Built target version_h
[  0%] Building C object src/CMakeFiles/libvgmstream.dir/coding/acm_decoder.c.o
...
[ 94%] Linking C static library libvgmstream.a
/Library/Developer/CommandLineTools/usr/bin/ranlib: file: libvgmstream.a(at3plus_decoder.c.o) has no symbols
/Library/Developer/CommandLineTools/usr/bin/ranlib: file: libvgmstream.a(mp4_aac_decoder.c.o) has no symbols
/Library/Developer/CommandLineTools/usr/bin/ranlib: file: libvgmstream.a(mp4_faac.c.o) has no symbols
/Library/Developer/CommandLineTools/usr/bin/ranlib: file: libvgmstream.a(at3plus_decoder.c.o) has no symbols
/Library/Developer/CommandLineTools/usr/bin/ranlib: file: libvgmstream.a(mp4_aac_decoder.c.o) has no symbols
/Library/Developer/CommandLineTools/usr/bin/ranlib: file: libvgmstream.a(mp4_faac.c.o) has no symbols
[ 94%] Built target libvgmstream
/bin/sh: ./autogen.sh: No such file or directory
make[2]: *** [CMakeFiles/CELT_0110_AUTOGEN] Error 127
make[1]: *** [CMakeFiles/CELT_0110_AUTOGEN.dir/all] Error 2
make: *** [all] Error 2
```

Homebrew's build also failed but has a different error: https://github.com/Homebrew/homebrew-core/pull/94247/files

**Comment by KatieFrogs** [comment]
Can you try removing the `dependencies` directory in the project root and try building again, I am guessing the file `celt-0.11.0.tar.gz` did not download correctly

**Comment by kode54** [comment]
The two .tar.gz files are downloading 0 bytes.

**Comment by kode54** [comment]
Lovely, downloads.xiph.org doesn't agree with curl on macOS:

```
% curl -I https://downloads.xiph.org/releases/celt/celt-0.6.1.tar.gz
curl: (35) error:1400442E:SSL routines:CONNECT_CR_SRVR_HELLO:tlsv1 alert protocol version
```

**Comment by xxyzz** [comment]
I have curl installed from Homebrew and it's in my PATH, I guess cmake only use macOS's builtin curl?

**Comment by kode54** [comment]
It uses /usr/lib/libcurl.4.dylib, from the prelinker cache.

**Comment by xxyzz** [comment]
cele build failed:

```
[ 94%] Built target libvgmstream
checking for autoconf...
-n checking for automake 1.6 or later...
automake
-n checking for aclocal 1.6 or later...
aclocal
-n checking for libtool...
glibtoolize
Generating configuration files for celt, please wait....
  aclocal
  autoheader
  glibtoolize --automake
  automake --add-missing
configure.ac:27: warning: AM_INIT_AUTOMAKE: two- and three-arguments forms are deprecated.  For more info, see:
configure.ac:27: https://www.gnu.org/software/automake/manual/automake.html#Modernize-AM_005fINIT_005fAUTOMAKE-invocation
configure.ac:31: installing './compile'
configure.ac:30: installing './config.guess'
configure.ac:30: installing './config.sub'
configure.ac:27: installing './install-sh'
configure.ac:27: installing './missing'
libcelt/Makefile.am:35: warning: 'INCLUDES' is the old name for 'AM_CPPFLAGS' (or '*_CPPFLAGS')
libcelt/Makefile.am: installing './depcomp'
parallel-tests: installing './test-driver'
tests/Makefile.am:1: warning: 'INCLUDES' is the old name for 'AM_CPPFLAGS' (or '*_CPPFLAGS')
tools/Makefile.am:8: warning: 'INCLUDES' is the old name for 'AM_CPPFLAGS' (or '*_CPPFLAGS')
  autoconf
configure.ac:5: warning: 'AM_CONFIG_HEADER': this macro is obsolete.
configure.ac:5: You should use the 'AC_CONFIG_HEADERS' macro instead.
./lib/autoconf/general.m4:2434: AC_DIAGNOSE is expanded from...
aclocal.m4:9922: AM_CONFIG_HEADER is expanded from...
configure.ac:5: the top level
configure.ac:27: warning: AM_INIT_AUTOMAKE: two- and three-arguments forms are deprecated.
./lib/autoconf/general.m4:2434: AC_DIAGNOSE is expanded from...
aclocal.m4:9570: AM_INIT_AUTOMAKE is expanded from...
configure.ac:27: the top level
configure.ac:31: warning: The macro `AM_PROG_LIBTOOL' is obsolete.
configure.ac:31: You should run autoupdate.
aclocal.m4:122: AM_PROG_LIBTOOL is expanded from...
configure.ac:31: the top level
configure.ac:33: warning: The macro `AC_PROG_CC_C99' is obsolete.
configure.ac:33: You should run autoupdate.
./lib/autoconf/c.m4:1659: AC_PROG_CC_C99 is expanded from...
configure.ac:33: the top level
configure.ac:42: warning: The macro `AC_TRY_COMPILE' is obsolete.
configure.ac:42: You should run autoupdate.
./lib/autoconf/general.m4:2847: AC_TRY_COMPILE is expanded from...
configure.ac:42: the top level
configure.ac:54: warning: The macro `AC_TRY_COMPILE' is obsolete.
configure.ac:54: You should run autoupdate.
./lib/autoconf/general.m4:2847: AC_TRY_COMPILE is expanded from...
configure.ac:54: the top level
configure.ac:70: warning: The macro `AC_HELP_STRING' is obsolete.
configure.ac:70: You should run autoupdate.
./lib/autoconf/general.m4:204: AC_HELP_STRING is expanded from...
aclocal.m4:9054: XIPH_PATH_OGG is expanded from...
lib/m4sugar/m4sh.m4:699: AS_IF is expanded from...
configure.ac:70: the top level
configure.ac:70: warning: The macro `AC_TRY_RUN' is obsolete.
configure.ac:70: You should run autoupdate.
./lib/autoconf/general.m4:2997: AC_TRY_RUN is expanded from...
aclocal.m4:9054: XIPH_PATH_OGG is expanded from...
lib/m4sugar/m4sh.m4:699: AS_IF is expanded from...
configure.ac:70: the top level
configure.ac:70: warning: The macro `AC_TRY_LINK' is obsolete.
configure.ac:70: You should run autoupdate.
./lib/autoconf/general.m4:2920: AC_TRY_LINK is expanded from...
aclocal.m4:9054: XIPH_PATH_OGG is expanded from...
lib/m4sugar/m4sh.m4:699: AS_IF is expanded from...
configure.ac:70: the top level
configure.ac:121: warning: The macro `AC_WARN' is obsolete.
configure.ac:121: You should run autoupdate.
./lib/autoconf/oldnames.m4:33: AC_WARN is expanded from...
lib/m4sugar/m4sh.m4:699: AS_IF is expanded from...
./lib/autoconf/general.m4:1534: AC_ARG_ENABLE is expanded from...
configure.ac:121: the top level
configure.ac:157: warning: AC_LANG_CONFTEST: no AC_LANG_SOURCE call detected in body
./lib/autoconf/lang.m4:199: AC_LANG_CONFTEST is expanded from...
./lib/autoconf/general.m4:2823: _AC_COMPILE_IFELSE is expanded from...
./lib/autoconf/general.m4:2839: AC_COMPILE_IFELSE is expanded from...
configure.ac:157: the top level
configure.ac:168: warning: The macro `AC_TRY_COMPILE' is obsolete.
configure.ac:168: You should run autoupdate.
./lib/autoconf/general.m4:2847: AC_TRY_COMPILE is expanded from...
lib/m4sugar/m4sh.m4:692: _AS_IF_ELSE is expanded from...
lib/m4sugar/m4sh.m4:699: AS_IF is expanded from...
./lib/autoconf/general.m4:2249: AC_CACHE_VAL is expanded from...
./lib/autoconf/general.m4:2270: AC_CACHE_CHECK is expanded from...
acinclude.m4:13: AC_C99_FUNC_LRINTF is expanded from...
configure.ac:168: the top level
configure.ac:169: warning: The macro `AC_TRY_COMPILE' is obsolete.
configure.ac:169: You should run autoupdate.
./lib/autoconf/general.m4:2847: AC_TRY_COMPILE is expanded from...
lib/m4sugar/m4sh.m4:692: _AS_IF_ELSE is expanded from...
lib/m4sugar/m4sh.m4:699: AS_IF is expanded from...
./lib/autoconf/general.m4:2249: AC_CACHE_VAL is expanded from...
./lib/autoconf/general.m4:2270: AC_CACHE_CHECK is expanded from...
acinclude.m4:29: AC_C99_FUNC_LRINT is expanded from...
configure.ac:169: the top level
configure.ac:205: warning: AC_OUTPUT should be used without arguments.
configure.ac:205: You should run autoupdate.
[ 94%] Built target CELT_0061_AUTOGEN
checking for a BSD-compatible install... /usr/local/bin/ginstall -c
checking whether build environment is sane... yes
checking for a race-free mkdir -p... /usr/local/bin/gmkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether to enable maintainer-specific portions of Makefiles... no
checking build system type... x86_64-apple-darwin21.3.0
checking host system type... x86_64-apple-darwin21.3.0
checking how to print strings... printf
checking whether make supports the include directive... yes (GNU style)
checking for gcc... /Library/Developer/CommandLineTools/usr/bin/cc
checking whether the C compiler works... no
configure: error: in `/Users/x/git/vgmstream/build/dependencies/celt-0061':
configure: error: C compiler cannot create executables
See `config.log' for more details
make[2]: *** [CMakeFiles/CELT_0061_CONFIGURE] Error 77
make[1]: *** [CMakeFiles/CELT_0061_CONFIGURE.dir/all] Error 2
make: *** [all] Error 2
```

`config.log`:

```
configure:4157: checking whether the C compiler works
configure:4179: /Library/Developer/CommandLineTools/usr/bin/cc -Dalg_quant=alg_0061_quant -Dalg_unquant=alg_0061_unquant -Dcelt_decode=celt_0061_decode -Dcelt_decoder_create=celt_0061_decoder_create -Dcelt_decoder_create_custom=celt_0061_decoder_create_custom -Dcelt_decoder_destroy=celt_0061_decoder_destroy -Dcelt_mode_create=celt_0061_mode_create -Dcelt_mode_destroy=celt_0061_mode_destroy -Dcelt_mode_info=celt_0061_mode_info -Dcelt_encoder_destroy=celt_0061_encoder_destroy -Dcelt_encoder_create=celt_0061_encoder_create -Dcelt_encode=celt_0061_encode -Dcelt_encode_float=celt_0061_encode_float -Dcelt_encoder_ctl=celt_0061_encoder_ctl -Dcelt_decode_float=celt_0061_decode_float -Dcelt_decoder_ctl=celt_0061_decoder_ctl -Dcompute_allocation=compute_0061_allocation -Dcompute_band_energies=compute_0061_band_energies -Ddenormalise_bands=denormalise_0061_bands -Dec_dec_init=ec_0061_dec_init -Dec_decode=ec_0061_decode -Dec_decode_bin=ec_0061_decode_bin -Dec_dec_update=ec_0061_dec_update -Dec_dec_uint=ec_0061_dec_uint -Dec_dec_bits=ec_0061_dec_bits -Dec_enc_init=ec_0061_enc_init -Dec_encode=ec_0061_encode -Dec_encode_bin=ec_0061_encode_bin -Dec_enc_uint=ec_0061_enc_uint -Dec_enc_bits=ec_0061_enc_bits -Dec_enc_done=ec_0061_enc_done -Dnormalise_bands=normalise_0061_bands -Drenormalise_vector=renormalise_0061_vector -Dquant_coarse_energy=quant_0061_coarse_energy -Dquant_fine_energy=quant_0061_fine_energy -Dquant_energy_finalise=quant_0061_energy_finalise -Dunquant_coarse_energy=unquant_0061_coarse_energy -Dunquant_energy_finalise=unquant_0061_energy_finalise -Dunquant_fine_energy=unquant_0061_fine_energy -DeMeans=eMeans_0061 -Dencode_pulses=encode_0061_pulses -Ddecode_pulses=decode_0061_pulses -Dec_laplace_encode=ec_0061_laplace_encode -Dec_laplace_decode=ec_0061_laplace_decode -Dcelt_decoder_get_size=celt_0061_decoder_get_size -Dcelt_decoder_init=celt_0061_decoder_init -Dec_dec_bit_logp=ec_0061_dec_bit_logp -Dec_dec_icdf=ec_0061_dec_icdf -Dec_enc_bit_logp=ec_0061_enc_bit_logp -Dec_enc_icdf=ec_0061_enc_icdf -Dec_enc_shrink=ec_0061_enc_shrink -D_celt_lpc=_celt_lpc_0061 -D_celt_autocorr=_celt_autocorr_0061 -Damp2Log2=amp2Log2_0061 -Dstereo_itheta=stereo_0061_itheta -Danti_collapse=anti_0061_collapse -Dspreading_decision=spreading_0061_decision -Dhaar1=haar1_0061 -Dquant_all_bands=quant_0061_all_bands -Dlog2_frac=log2_0061_frac -Dicwrs=icwrs_0061 -Dget_required_bits=get_0061_required_bits -Dec_ilog=ec_0061_ilog -Dmdct_forward=mdct_0061_forward -Dmdct_backward=mdct_0061_backward -Dmdct_init=mdct_0061_init -Dmdct_clear=mdct_0061_clear -fPIC   conftest.c  >&5
ld: library not found for -lSystem
clang: error: linker command failed with exit code 1 (use -v to see invocation)
configure:4183: $? = 1
configure:4223: result: no
configure: failed program was:
| /* confdefs.h */
| #define PACKAGE_NAME ""
| #define PACKAGE_TARNAME ""
| #define PACKAGE_VERSION ""
| #define PACKAGE_STRING ""
| #define PACKAGE_BUGREPORT ""
| #define PACKAGE_URL ""
| /* end confdefs.h.  */
| 
| int
| main (void)
| {
| 
|   ;
|   return 0;
| }
```

**Comment by kode54** [comment]
I literally have no idea why CMake is invoking the compiler using direct path to /Library/Developer/CommandLineTools/usr/bin/cc or $Xcode.app/Contents/Developer/etc, but it breaks the compiler's ability to search for its own headers. CMake should be invoking the compiler using /usr/bin/cc.

**Comment by kode54** [comment]
I pushed a dodgy fix, f9230158d953dfb0fb14a2a57052a8460a5d43dd - You'll still need to `export LIBRARY_PATH=` pointing to your Homebrew libraries folder, depending either aarch64 or x86_64, for it to pick up libao and libmpg123.

**Comment by xxyzz** [comment]
Now the compilation is successful. Thanks for your help!
